browser/test/integration/loadAllPlugins.js

```diff
@@ -0,0 +1,284 @@
+THREE.PerspectiveCamera.prototype = Object.create(THREE.Object3D.prototype)
+
+global.WebVRManager = function(){}
+global.THREE.Matrix4.prototype.identity = function() {}
+
+global.UndoManager = require('../../scripts/commands/undoManager.js')
+global.GraphApi = require('../../scripts/graphApi.js')
+global.Connection = require('../../scripts/connection.js').Connection
+global.ConnectionUI = require('../../scripts/connection.js').ConnectionUI
+global.ConnectionUI.prototype.resolve_slot_divs = function() {
+	this.src_slot_div = $()
+	this.dst_slot_div = $()
+}
+global.navigator = { userAgent: 'test' }
+global.mixpanel = { track: function() {} }
+
+var pluginPath = './browser/plugins/'
+
+describe('loadAllPlugins', function() {
+
+	var Core = require('../../scripts/core')
+
+	function PluginManager() {
+		EventEmitter.call(this)
+		this.keybyid = {}
+		var that = this
+		process.nextTick(function() {
+			that.emit('ready')
+		})
+	}
+	PluginManager.prototype = Object.create(EventEmitter.prototype)
+	PluginManager.prototype.create = function(id, node) {
+		if (!E2.plugins[id])
+			loadPlugin(id)
+		console.log('load', id, !!E2.plugins[id])
+		var p = new E2.plugins[id](core, node)
+		p.id = id
+		return p
+	}
+
+	global.PluginManager = PluginManager
+
+	// global.Node.prototype.initialise = function(){}
+
+	function TreeNode() {}
+	TreeNode.prototype.add_child = function() { return new TreeNode() }
+	global.TreeNode = TreeNode
+
+	beforeEach(function() {
+		global.window = { location: { pathname: 'test/test' } }
+		var dummyCore = reset()
+		E2.commands.graph = require('../../scripts/commands/graphEditCommands')
+		E2.EnvironmentSettings = function(){}
+		E2.Noise = function() {this.noise2D = function(){}}
+
+		require('../../scripts/variables')
+		require('../../scripts/util')
+
+		app = E2.app = new Application()
+		app.updateCanvas = function() {}
+		core = E2.core = new Core()
+		core.renderer = dummyCore.renderer
+		E2.app.player = { core: core }
+		core.active_graph = new Graph(core, null, new TreeNode())
+		core.root_graph = core.active_graph
+		core.graphs = [ core.active_graph ]
+		core.renderer = {setPixelRatio: function(){}}
+
+		E2.dom = E2.dom || {}
+		E2.dom.webgl_canvas = {
+			on:function(){},
+			bind:function(){},
+			'0': {
+				clientWidth:1,
+				clientHeight:1
+			}
+		}
+
+		// throw any errors
+		global.msg = function(txt) {
+			if (/^ERROR/.test(txt))
+				throw new Error(txt)
+		}
+
+	})
+
+	var exceptions = [
+		// legacy plugins used in unit tests (paste_complex)
+		'mesh_renderer_emitter',
+		'perspective_camera',
+		'grid_mesh_generator',
+		'concatenate_matrix_modulator',
+		'scene_renderer_emitter',
+
+		// these will need to be fixed:
+		'viewport_height_generator',
+		'viewport_width_generator',
+	]
+
+	it('loads all plugins', function(done) {
+		fs.readdir(pluginPath, function(err, files) {
+			if (err) {
+				console.error(err);
+				return
+			}
+
+			for(var i = 0; i < files.length; ++i) {
```

Comment:
might want to have an assertion or two here, perhaps a https://nodejs.org/api/assert.html#assert_assert_doesnotthrow_block_error_message

as well as a number of plugins actually loaded, for example, just to ensure this actually does something :)